python/paddle/distributed/passes/auto_parallel_c_embedding.py

```diff
@@ -0,0 +1,357 @@
+                        stack.append(next_op)
+                        pre_id = op.id()
+
+                    process_mesh = (
+                        op.results()[0].process_mesh
+                        if op.num_results() > 0
+                        else op.operand(0).source().process_mesh
+                    )
+                    op.dist_attr = (
+                        paddle.base.libpaddle.pir.create_op_dist_attribute(
+                            process_mesh,
+                            operands,
+                            results,
+                        )
+                    )
+
+
+def update_startup_program(startup_program, mp_axis):
+    startup_block = startup_program.global_block()
+    for op in startup_block.ops:
+        if op.name() == 'pd_op.full':
+            next_op = op.result(0).all_used_ops()[0]
+            parameter_name = next_op.str_attr("parameter_name")
+            pattern = re.compile(r'embedding_.*\.w_0\.dist')
+            if pattern.match(parameter_name):
+                placements = op.results()[0].placements
+                dim_map, partial_status = (
+                    dist.auto_parallel.placement_type.to_dim_map(
+                        placements, len(placements)
+                    )
+                )
+                dim_map = [mp_axis, -1]
+                dist_attr = (
+                    paddle.base.libpaddle.pir.create_tensor_dist_attribute(
+                        op.results()[0].process_mesh, dim_map, partial_status
+                    )
+                )
+                dist_type = paddle.base.libpaddle.pir.cvt_to_dist_type(
+                    op.results()[0].type(), dist_attr
+                )
+                op.results()[0].set_type(dist_type)
+                op.dist_attr = (
+                    paddle.base.libpaddle.pir.create_op_dist_attribute(
+                        op.results()[0].process_mesh, [], [dist_attr]
+                    )
+                )
+
+
+@register_pass("auto_parallel_c_embedding_pass")
+class AutoParallelCEmbeddingPass(PassBase):
+    def __init__(self):
+        super().__init__()
+
+    def _check_self(self):
+        return True
+        hcg = fleet.get_hybrid_communicate_group()
+        mp_size = hcg.get_model_parallel_world_size()
+        if mp_size > 1:
+            return True
+        warnings.warn(
+            "The c_embedding pass is only applicable to tnesor parallel."
+        )
+        return False
+
+    def _check_conflict(self, other_pass):
+        return True
+
+    def _apply_single_impl(self, main_program, startup_program, context):
+        concrete_program = self.get_attr("concrete_program")
+        ops = main_program.global_block().ops
+        for i, op in enumerate(ops):
+            if op.name() == 'pd_op.embedding':
+                # update weight dims mapping
+                mp_axis = update_weight(op, concrete_program)
+
+                # update startup_program
```

Comment:
Meaningless comments.   delete line 334, 347, 350, 353...

Reply:
Thx, the code has been modified.